gtsam/discrete/AlgebraicDecisionTree.h

```diff
@@ -57,6 +57,9 @@ namespace gtsam {
 
     AlgebraicDecisionTree(double leaf = 1.0) : Base(leaf) {}
 
+    /// Constructor which accepts root pointer
+    AlgebraicDecisionTree(const typename Base::NodePtr root) : Base(root) {}
```

Comment:
I added this new constructor so I could set the ADT root node to null explicitly.